include/beman/execution26/detail/unspecified_promise.hpp

```diff
@@ -0,0 +1,20 @@
+// include/beman/execution26/detail/unspecified_promise.hpp           -*-C++-*-
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef INCLUDED_BEMAN_EXECUTION26_DETAIL_UNSPECIFIED_PROMISE
+#define INCLUDED_BEMAN_EXECUTION26_DETAIL_UNSPECIFIED_PROMISE
+
+#include <coroutine>
+
+namespace beman::execution26::detail {
+struct unspecified_promise {
+    auto get_return_object() noexcept -> unspecified_promise;
+    auto initial_suspend() noexcept -> unspecified_promise;
+    auto final_suspend() noexcept -> unspecified_promise;
```

Comment:
This promise type isn't supposed to be used. However, I _think_ it would be reasonable to declare it in a form such that it _could_ be a promise type if it were implemented. To that end I think `initial_suspend` and `final_suspend` should return an awaiter, e.g. `std::suspend_never`.

Reply:
changed